src/components/GlobalPreferences/GlobalPreferences.js

```diff
@@ -131,14 +144,21 @@ function useGlobalPreferences(locator = {}) {
 
   useEffect(() => {
     const { preferences: { path = '' } = {} } = locator
-    if (!path || !SECTIONS.has(path)) {
+    if (!path) {
       setSectionIndex(null)
       return
     }
-    setSectionIndex(PATHS.findIndex(item => item === path))
-  }, [locator])
+    const index = PATHS.findIndex(item => path.startsWith(item))
+    setSectionIndex(index === -1 ? null : index)
+
+    // subsection is the part after the PATH, e.g. for `?p=/notifications/verify` - `/verify`
+    const subsection = index !== -1 ? path.substring(PATHS[index].length) : null
```

Comment:
```suggestion
    const [, subsection = null] = index !== -1 ? PATHS[index].split('/') : ''
```

Reply:
The subsection is derived from the current path. Splitting `PATHS[index]` doesn't take the current path into account

Reply:
Oh, true sorry about suggesting something erroneous, but you get the gist, just an idea to use split + array destructuring, nothing else